### PR TITLE
Build dethrace with -Wstrict-prototypes

### PR DESCRIPTION
**Pull request by madebr**
`-Wstrict-prototypes`  will warn when functions are declared as:
```c
void my_function();
```
This function is compatible with the following function definition:
```c
void my_function(int arg1, int arg2) {
} 
```
This should hopefully avoid the class of errors fixed by https://github.com/dethrace-labs/dethrace/pull/313.

After this pr, the following warning is emitted:
```
[114/210] Building C object src/harness/CMakeFiles/harness.dir/platforms/sdl_opengl.c.o
/home/maarten/projects/dethrace/src/harness/platforms/sdl_opengl.c: In function ‘Harness_Platform_Init’:
/home/maarten/projects/dethrace/src/harness/platforms/sdl_opengl.c:243:37: warning: assignment to ‘void (*)(void)’ from incompatible pointer type ‘void (*)(tRenderer_flush_type)’ [-Wincompatible-pointer-types]
  243 |     platform->Renderer_FlushBuffers = GLRenderer_FlushBuffers;
      |                         
```

Also fixed by this pr is a warnings emitted around non-returning functions:
```
[114/171] Building C object src/DETHRACE/CMakeFiles/dethrace_obj.dir/common/errors.c.o
/home/maarten/projects/dethrace/src/DETHRACE/common/errors.c: In function ‘FatalError’:
/home/maarten/projects/dethrace/src/DETHRACE/common/errors.c:173:1: warning: ‘noreturn’ function does return
  173 | }
      | ^
```

The last commit separates `null.h` in a header and implementation file.

### CODE REVIEW

File: src/DETHRACE/pc-dos/dosnet.c [hunk 1]
```diff
@@ -36,7 +36,33 @@ unsigned short gECB_offset;
...
+/*static*/ int hmiIPXSendDataDirect(char* pHeader, W32 wHSize, char* pData, W32 wDSize, _NETNOW_NODE_ADDR* sNode);
+/*static*/ void hmiIPXGetInternetworkAddr(_IPX_INTERNET_ADDR* sInterworkAddr);
+/*static*/ void hmiIPXGetLocalTarget(_IPX_LOCAL_TARGET* sNetworkAddr);
+/*static*/ int AllocateRealMem(W32 wSize, char** pPtr, W32* pSegment, tU16* pSelector);
+/*static*/ int FreeRealMem(tU16 pSelector);
+/*static*/ int hmiIPXInstalled(void);
+/*static*/ void GetLargestPacketSizeOoErBetterInsertLinfordChristieJokeHere(void);
+/*static*/ void GetIPXToStickItsEarToTheGround(void);
+
+    // IDA: void __cdecl ClearupPDNetworkStuff()
```
Comment (madebr) [inline src/DETHRACE/pc-dos/dosnet.c:65]:
```suggestion
// IDA: void __cdecl ClearupPDNetworkStuff()
```

File: src/DETHRACE/pc-win95/win95sys.c [hunk 2]
```diff
@@ -77,17 +77,37 @@ int gNetwork_profile_file_exists = 0;
...
+static void Usage(char* pProgpath);
+/*static*/ int OurGetChar(void);
+/*static*/ int InitJoysticks(void);
+/*static*/ tU32 ReadJoystickAxis(int pBit);
+static void Win32BRenderWarningFunc(char* msg);
+static void Win32BRenderFailureFunc(char* msg);
+
+
+
+
```
Comment (madebr) [inline src/DETHRACE/pc-win95/win95sys.c:109]:
```suggestion

```

File: src/DETHRACE/pc-win95/win95sys.c [hunk 3]
```diff
@@ -77,17 +77,37 @@ int gNetwork_profile_file_exists = 0;
...
+extern void QuitGame(void);
```
Comment (madebr) [inline src/DETHRACE/pc-win95/win95sys.c:110]:
```suggestion
static void Win32BRenderFailureFunc(char* msg);

extern void QuitGame(void);
```

**Review by dethrace-labs — COMMENTED** [review]
LGTM!

**Comment by madebr** [comment]
You need to approve before I can merge this pr :)

**Review by dethrace-labs — APPROVED** [review]

**Comment by dethrace-labs** [comment]
Oops, thought I had!

**Pull request merged**